### ISSUE TITLE
[Clang-Tidy] readability-use-std-min-max should consider `std::clamp`

### ISSUE THREAD

**Issue by SunBlack** [llvm/llvm-project#97835]
Tried the new check `readability-use-std-min-max` of the upcoming Clang-Tidy 19.

In one case we had following code:
```cpp
if (ratio < 0.0F)
{
	ratio = 0.0F;
}
if (ratio > 1.0F)
{
	ratio = 1.0F;
}
```

Running `run-clang-tidy-19  -header-filter='.*' -checks='-*,readability-use-std-min-max' -fix` results in:
```cpp
ratio = std::max(ratio, 0.0F);
ratio = std::min(ratio, 1.0F);
```

But in this case it would be better
```cpp
ratio = std::clamp(ratio, 0.0F, 1.0F);
```

I don't know how complex it would be to recognize this pattern, i.e. whether it is better to introduce a `readability-use-std-clamp`, even if this may require a second pass, or whether this can be combined by changing the check to `readability-use-std-min-max-clamp`.

**Comment by mathstuf** [comment]
Related (on the detection side) is the pattern of:

```c++
out = std::max(in[0], out);
out = std::max(in[1], out);
out = std::max(in[2], out);
```

which could be:

```c++
out = std::max({ in[0], in[1], in[2] });
```

**Comment by sparr** [comment]
I would also like to see this check suggest replacing `a = std::min( b, std::max( c, a ) )` (and other similar constructions) with `a = std::clamp( a, c, b )`

**Comment by sparr** [comment]
Note that `std::min( a, std::max( b, c ) )` without an assignment or other reference to one of the three variables can't be simplified, because we don't know which of `b` or `c` is the value being clamped and which is the lower bound.

**Comment by carlosgalvezp** [comment]
> Related (on the detection side) is the pattern of:
> 
> out = std::max(in[0], out);
> out = std::max(in[1], out);
> out = std::max(in[2], out);
> which could be:
> 
> out = std::max({ in[0], in[1], in[2] });

This is already solved by ``

**Comment by SunBlack** [comment]
@carlosgalvezp This issue wasn't about initializer list variant. It was about using `std::clamp`

**Comment by carlosgalvezp** [comment]
> [@carlosgalvezp](https://github.com/carlosgalvezp) This issue wasn't about initializer list variant. It was about using `std::clamp`

Yes sorry, I must have missclicked on the phone, the issue is relevant.

**Comment by carlosgalvezp** [comment]
For `std::clamp` we probably could have a `modernize-use-std-clamp`.
